backend/common_utils/src/dapr.rs

```diff
@@ -0,0 +1,334 @@
+            error_code,
+            error_reason,
+            request_data,
+            response_data,
+            additional_fields,
+        }
+    }
+}
+
+/// Configuration for events system
+#[derive(Debug, Clone, Deserialize)]
+pub struct EventConfig {
+    pub enabled: bool,
+    pub pubsub_component: String,
+    pub topic: String,
+    #[serde(default)]
+    pub transformations: HashMap<String, String>, // target_path → source_field
+    #[serde(default)]
+    pub static_values: HashMap<String, String>, // target_path → static_value
+    #[serde(default)]
+    pub extractions: HashMap<String, String>, // target_path → extraction_path
+}
+
+impl Default for EventConfig {
+    fn default() -> Self {
+        Self {
+            enabled: false,
+            pubsub_component: "kafka-pubsub".to_string(),
+            topic: "connector-events".to_string(),
+            transformations: HashMap::new(),
+            static_values: HashMap::new(),
+            extractions: HashMap::new(),
+        }
+    }
+}
+
+/// Context data available for event processing
+#[derive(Debug, Clone)]
+pub struct EventContext {
+    pub request_data: Option<serde_json::Value>,
+}
+
+impl Default for EventContext {
+    fn default() -> Self {
+        Self { request_data: None }
+    }
+}
+
+// Define FlowName enum locally to avoid circular dependency
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+pub enum FlowName {
+    Authorize,
+    Refund,
+    Capture,
+    Void,
+    Psync,
+    Rsync,
+    AcceptDispute,
+    SubmitEvidence,
+    DefendDispute,
+    Dsync,
+    IncomingWebhook,
+    SetupMandate,
+    CreateOrder,
+}
+
+impl FlowName {
+    pub fn as_str(&self) -> &'static str {
+        match self {
+            Self::Authorize => "Authorize",
+            Self::Refund => "Refund",
+            Self::Capture => "Capture",
+            Self::Void => "Void",
+            Self::Psync => "Psync",
+            Self::Rsync => "Rsync",
+            Self::AcceptDispute => "AcceptDispute",
+            Self::SubmitEvidence => "SubmitEvidence",
+            Self::DefendDispute => "DefendDispute",
+            Self::Dsync => "Dsync",
+            Self::IncomingWebhook => "IncomingWebhook",
+            Self::SetupMandate => "SetupMandate",
+            Self::CreateOrder => "CreateOrder",
+        }
+    }
+}
+
+#[derive(Debug, Clone)]
+pub enum EventStage {
+    RequestReceived,
+    RequestSent,
+    ResponseReceived,
+}
+
+impl EventStage {
+    pub fn as_str(&self) -> &'static str {
+        match self {
+            Self::RequestReceived => "RequestReceived",
+            Self::RequestSent => "RequestSent",
+            Self::ResponseReceived => "ResponseReceived",
+        }
+    }
+}
+
+/// Create a Dapr client connection
+pub async fn create_client() -> Result<Client<dapr::client::TonicClient>> {
+    let dapr_port = std::env::var("DAPR_GRPC_PORT").unwrap_or_else(|_| "50001".to_string());
+    let addr = format!("http://localhost:{dapr_port}");
```

Comment:
- why localhost is hardcoded here, any specific reason?
- why we reading env directly by using std::env::var? (there is state based approach on reading env exists Check Config type for reference and how this is used in payment related flow)

Reply:
Updated.